pytorch_lightning/core/datamodule.py

```diff
@@ -38,6 +39,13 @@ def __call__(cls, *args, **kwargs):
         2. Assures prepare_data() runs on rank 0
         3. Lets you check prepare_data and setup to see if they've been called
         """
+        __flash_special_attr__ = getattr(cls, "__flash_special_attr__", None)
```

Comment:
this feels too magical as a hack to make flash work. can we make this more explicit in the interface for what these attributes are?

Reply:
Yes, it is used to transform class attributes into instance level attributes through a deepcopy. I didn't find a better solution yet. Any idea ?

Reply:
Inherit the object in Flash and add this logic there? This should be kept in flash imo

Reply:
Definitely worth to explore, but this is part of the DataModule meta class.Not sure how subclassing metaclass works.

Reply:
Moved back to Flash.